ilastik/applets/dataExport/dataExportGui.py

```diff
@@ -223,11 +227,86 @@ def initCentralUic(self):
 
         # Set up handlers
         self.batchOutputTableWidget.itemSelectionChanged.connect(self.handleTableSelectionChange)
+        # Handle the right-click event
+        self.batchOutputTableWidget.setContextMenuPolicy(Qt.CustomContextMenu)
+        self.batchOutputTableWidget.customContextMenuRequested.connect(self.showContextMenu)
 
         # Set up the viewer area
         self.initViewerStack()
         self.splitter.setSizes([150, 850])
 
+    # List of all valid file extensions that ilastik might export.
+    VALID_EXPORT_EXTENSIONS = {
+        ".tiff",
+        ".tif",
+        ".h5",
+        ".png",
+        ".jpg",
+        ".jpeg",
+        ".bmp",
+        ".npy",
+        ".csv",
+        ".tsv",
+        ".xlsx",
+        ".hdf5",
+    }
+
+    def showContextMenu(self, pos):
+        """Handles right-click menu in the export file list."""
+        menu = QMenu(self)
+
+        # Determine which cell in the table was clicked
+        row = self.batchOutputTableWidget.rowAt(pos.y())
+        col = self.batchOutputTableWidget.columnAt(pos.x())
+
+        # Only show the option if the user clicked in the 'Export Location' column
+        if col == Column.ExportLocation:
+            openFolderAction = menu.addAction("Open Export Folder")
+            openFolderAction.triggered.connect(lambda: self.openExportFolder(row))
+
+        # Show the context menu at the correct position
+        menu.exec_(self.batchOutputTableWidget.mapToGlobal(pos))
+
+    def openExportFolder(self, row):
+        """Opens the parent folder of the exported file, handling all supported formats."""
+        # Get the export path from the selected row
+        exportPath = self.batchOutputTableWidget.item(row, Column.ExportLocation).text()
+
+        if not exportPath:
```

Comment:
Maybe rather show the context menu with the "Open Export Folder" option grayed out. This would require to get the data already in `showContextMenu`.